packages/sdk/src/ParadymWalletSdk.tsx

```diff
@@ -22,32 +23,82 @@ import { type InvitationResult, parseDidCommInvitation, parseInvitationUrl } fro
 import { type ResolveOutOfBandInvitationResult, resolveOutOfBandInvitation } from './invitation/resolver'
 import { AgentProvider, useAgent } from './providers/AgentProvider'
 import { type CredentialRecord, deleteCredential, storeCredential } from './storage/credentials'
+import type { TrustMechanismConfiguration } from './trust/trustMechanism'
 
 export type ParadymWalletSdkResult<T extends Record<string, unknown> = Record<string, unknown>> =
   | ({ success: true } & T)
   | { success: false; message: string }
 
-export type ParadymWalletSdkOptions = SetupAgentOptions
+export type ParadymWalletSdkOptions = SetupAgentOptions & {
+  /**
+   *
+   * Trust mechanisms supported by the wallet
+   *
+   * The order matters. The first index will be tried first, until the last
+   *
+   * When one is found that works, it will be used
+   *
+   */
+  // TODO(sdk): this will get more complex, as eudi_rp_auth needs more configuration
+  trustMechanisms: TrustMechanismConfiguration[]
+}
 
 export type SetupParadymWalletSdkOptions = Omit<ParadymWalletSdkOptions, 'id' | 'key'>
 
 export class ParadymWalletSdk {
+  public trustMechanisms: TrustMechanismConfiguration[]
   public readonly agent: FullAgent
 
   public constructor(options: ParadymWalletSdkOptions) {
     this.agent = setupAgent(options) as FullAgent
+    this.trustMechanisms = options.trustMechanisms
   }
 
   private assertAgentIsInitialized() {
     if (!this.agent.isInitialized) throw new ParadymWalletMustBeInitializedError()
   }
 
+  public get walletId() {
+    const walletKeyVersion = secureWalletKey.getWalletKeyVersion()
+
+    // TODO(sdk): how do we want to deal with the `easypid` wallet id?
+    return `easypid-wallet-${walletKeyVersion}`
+  }
+
+  public async reset(additionalResetCb?: () => Promise<void> | void) {
```

Comment:
I'm never too sure what a callback that is called at the end adds. 

It's easier IMO to just do:
```
sdk.reset()
myExtraResetSteps()
```

Reply:
Yeah it was left over from when something happened afterwards, but can be removed again.